### PR TITLE
updated libpd and test_libpd

### PR DESCRIPTION
**Pull request by danomatika**
This PR updates the newly-included libpd example project:

* rename makefiles into standard "Makefile" so building and running for beginners is easier
* updated libpd Makefile with windows and mac support from the libpd repo's Makefile
* added libpd staticlib build by default (in addition to dynamic lib)
* moved test_libpd into it's own subdirectory so it is better encapsulated with it's own Makefile
* integrated libpd directory into the autotools build so it is included with the source distribution tarball generated via: `make dist`

### CODE REVIEW

**Comment by danomatika** [comment]
The `libpd/Makefile` was test on macOS and hopefully works on Windows too.

@Spacechild1 @Lucarda If either of y'all have time, can you try building?

    git checkout update/libpd-example
    cd libpd
    make
    cd test_libpd
    make
    ./test_libpd

**Comment by Lucarda** [comment]
> cd libpd
> make

$ make
make: *** No targets specified and no makefile found.  Stop.

**Comment by danomatika** [comment]
What are there folder contents? Is the Makefile there?

**Comment by Lucarda** [comment]
nope.

see "file changes" in this PR. "the file has been deleted"

**Comment by danomatika** [comment]
Dammit. I have to make it all over again.

**Comment by Lucarda** [comment]
:)

**Comment by danomatika** [comment]
Thanks for the sympathy 🙄 . Please pull and try again...

**Comment by Lucarda** [comment]
Strange. Looks like the LDFLAGS is not getting `-lws2_32` ?

````
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x324): undefined reference to `__imp_select'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x35c): undefined reference to `__WSAFDIsSet'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0xfe3): undefined reference to `__imp_send'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1348): undefined reference to `__imp_socket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x13cb): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1597): undefined reference to `__imp_socket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x160c): undefined reference to `__imp_bind'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1637): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1751): undefined reference to `__imp_listen'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1767): undefined reference to `__imp_accept'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x17e6): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1880): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x18c3): undefined reference to `gai_strerrorA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x199b): undefined reference to `gai_strerrorA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x1b35): undefined reference to `__imp_send'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x2194): undefined reference to `__imp_recv'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x21ac): undefined reference to `__imp_getpeername'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_inter.o:s_inter.c:(.text+0x22c1): undefined reference to `__imp_recvfrom'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x10f): undefined reference to `__imp_WSAAddressToStringA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x1da): undefined reference to `__imp_getaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x376): undefined reference to `__imp_WSAAddressToStringA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x37d): undefined reference to `__imp_ntohs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x485): undefined reference to `__imp_ntohs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x515): undefined reference to `__imp_WSAAddressToStringA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x596): undefined reference to `__imp_ntohs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x5d5): undefined reference to `__imp_htons'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x62d): undefined reference to `__imp_ntohl'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x67c): undefined reference to `__imp_WSAStartup'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x6ba): undefined reference to `__imp_ioctlsocket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x6f5): undefined reference to `__imp_connect'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x732): undefined reference to `__imp_WSAGetLastError'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x7bd): undefined reference to `__imp_select'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x7d2): undefined reference to `__WSAFDIsSet'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x805): undefined reference to `__imp_getsockopt'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x80f): undefined reference to `__imp_WSASetLastError'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x843): undefined reference to `__imp_WSASetLastError'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x876): undefined reference to `__imp_closesocket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x89e): undefined reference to `__imp_getsockname'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x8cf): undefined reference to `__imp_ntohs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x8fb): undefined reference to `__imp_setsockopt'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x927): undefined reference to `__imp_ioctlsocket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x95b): undefined reference to `__imp_ioctlsocket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0x9bc): undefined reference to `__imp_setsockopt'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0xa07): undefined reference to `__imp_setsockopt'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0xa4c): undefined reference to `__imp_setsockopt'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0xa97): undefined reference to `__imp_setsockopt'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0xaa8): undefined reference to `__imp_WSAGetLastError'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: s_net.o:s_net.c:(.text+0xaf8): undefined reference to `__imp_WSAGetLastError'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x184): undefined reference to `__imp_send'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x18b): undefined reference to `__imp_sendto'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x4aa): undefined reference to `__imp_socket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x557): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x62b): undefined reference to `gai_strerrorA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x66c): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x6be): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x73b): undefined reference to `__imp_bind'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x76a): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x847): undefined reference to `gai_strerrorA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x868): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x8c0): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x107a): undefined reference to `__imp_socket'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x1120): undefined reference to `__imp_bind'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x112c): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x129c): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x132c): undefined reference to `__imp_listen'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x1365): undefined reference to `__imp_bind'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x1374): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x13ba): undefined reference to `__imp_freeaddrinfo'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x1473): undefined reference to `gai_strerrorA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x1518): undefined reference to `gai_strerrorA'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x1566): undefined reference to `__imp_accept'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x27bd): undefined reference to `__imp_recv'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x2b07): undefined reference to `__imp_recvfrom'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: x_net.o:x_net.c:(.text+0x2c0c): undefined reference to `__imp_getpeername'
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:116: pd.dll] Error 1

````

**Comment by danomatika** [comment]
Try changing:

```Makefile
    LIBPD_IMPLIB = libs/libpd.lib
     LIBPD_DEF = libs/libpd.def
```

to

```Makefile
    LIBPD_IMPLIB = libpd.lib
     LIBPD_DEF = libpd.def
```

**Comment by Lucarda** [comment]
same as above.

**Comment by Lucarda** [comment]
I changed

````
$(LIBPD): $(OBJ)
	$(CC) -o $(LIBPD) $(OBJ) -shared -lm -lpthread 
````
to

````
$(LIBPD): $(OBJ)
	$(CC) -o $(LIBPD) $(OBJ) -shared -lm -lpthread $(PLATFORM_LDFLAGS)
````

and it finish the compile (further tests on my side needed)

**Comment by Lucarda** [comment]
```
$ cd test_libpd
$ make
cc -I../../src -O3   -c -o test_libpd.o test_libpd.c
make: *** No rule to make target '../libpd.dll', needed by 'test_libpd'.  Stop.
```


seems I got one '../pd.dll' instead of '../libpd.dll'

**Comment by danomatika** [comment]
Ok. More fixes pushed now. If it fails to fid pd.dll now, then it probably has to be copied next to text_libpd.exe like on macOS.

**Comment by Lucarda** [comment]
I had to copy `pd.dll` and get `libwinpthread-1.dll` to the folder. But I got:


````
pd 0.52.0test2
bonk version 1.5
fiddle version 1.1 TEST4
pd~ version 0.54
pique 0.1 for PD version 23
sigmund~ version 0.07
1003-frequency: 480
1003-frequency: -480
instance 1, tick 0:
0.000000 0.000000 0.010884 0.010884 0.021769 0.021769 0.032653 0.032653 ...
instance 2, tick 0:
0.000000 0.000000 0.990000 0.990000 0.980000 0.980000 0.970000 0.970000 ...
print: 0
instance 1, tick 1:
0.696599 0.696599 0.707483 0.707483 0.718367 0.718367 0.729252 0.729252 ...
print: 0
instance 2, tick 1:
0.360000 0.360000 0.350000 0.350000 0.340000 0.340000 0.330000 0.330000 ...
print: 1
instance 1, tick 2:
0.393197 0.393197 0.404082 0.404082 0.414966 0.414966 0.425850 0.425850 ...
print: 1
instance 2, tick 2:
0.720000 0.720000 0.710000 0.710000 0.700000 0.700000 0.690000 0.690000 ...

````

**Comment by danomatika** [comment]
Cool. Thanks. I will add the copy to the makefile. Does mingw2/msys2 ship with a winpthread dll yet?

**Comment by Lucarda** [comment]
> Does mingw2/msys2 ship with a winpthread dll yet?

yes.

this is from the makefile:
````
$ make
cc -I../../src -O3   -c -o test_libpd.o test_libpd.c
cc -o test_libpd test_libpd.o ../pd.dll
if [ windows = "mac" ]; then cp ../pd.dll ./; fi

````

**Comment by Lucarda** [comment]
see https://github.com/porres/pd-cyclone/blob/773f6ac23d591b347763edc946dd995663ca2f65/Makefile#L311

on how to get the `pthread.dll`

**Comment by danomatika** [comment]
Ok, latest commit adds simple winpthread copy using the MINGW_PREFIX. I don't think we need/want to get much more complicated.

**Comment by Lucarda** [comment]
Wait:

````
$ pure-data/libpd/test_libpd
$ make
Makefile:47: *** empty variable name.  Stop.
````

I'll try to get a solution

**Comment by Lucarda** [comment]
Ok, just change [line](https://github.com/pure-data/pure-data/blob/ff493c13639bfecbfa5f0a5b76627d780594ff60/libpd/test_libpd/Makefile#L47):


````
$(PDTHREAD_DIR) = ${MINGW_PREFIX}/bin
$(PTHREAD) = libwinpthread-1.dll

$(PTHREAD):
	cp $(PTHREAD_DIR))/$(PTHREAD) .
````

to

````
PTHREAD_DIR = ${MINGW_PREFIX}/bin
PTHREAD = libwinpthread-1.dll

$(PTHREAD):
	cp $(PTHREAD_DIR)/$(PTHREAD) .

````

is just that and its done.

**Comment by danomatika** [comment]
Ah, duh, thanks. Got it. Thanks for testing and I can add the ein-thread solution to the libpd repo Makefile too.

@millerpuckette @umlaeute This PR is working for macOS and Windows. You can try it on Linux.

File: libpd/test_libpd/Makefile [hunk 1]
```diff
@@ -44,8 +44,8 @@ $(LIBPD):
 # on windowes, copy libpd and MinGW winpthread dll to here
 ifeq ($(PLATFORM), windows)
 
-$(PDTHREAD_DIR) = ${MINGW_PREFIX}/bin
-$(PTHREAD) = libwinpthread-1.dll
+PDTHREAD_DIR = ${MINGW_PREFIX}/bin
```
[the remark this answers is not in the document]
Comment (danomatika) [reply libpd/test_libpd/Makefile]: Doh. Fixed in latest commit.

File: libpd/test_libpd/Makefile [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	cp $(LIBPD_DIR)/$(LIBPD) .
+
+# on windowes, copy libpd and MinGW winpthread dll to here
+ifeq ($(PLATFORM), windows)
+
+PTHREAD_DIR = ${MINGW_PREFIX}/bin
+PTHREAD = libwinpthread-1.dll
+
+$(PTHREAD):
+	cp $(PTHREAD_DIR))/$(PTHREAD) .
```
Comment (Lucarda) [inline libpd/test_libpd/Makefile:51]: here there's another typo (etrxtra ")")

change to:
`cp $(PTHREAD_DIR)/$(PTHREAD) . `

Comment (danomatika) [reply libpd/test_libpd/Makefile]: Ok, maybe I got it now. Thanks for the "debug by proxy" help.

**Pull request merged**